scrc/dataset_creation/dataset_creator.py

```diff
@@ -652,6 +655,30 @@ def save_dataset(self, dataset: datasets.Dataset, labels: list, folder: Path,
         self.logger.info(f"Saved dataset files to {folder}")
         return splits
 
+    def clean_dataset(self, dataset):
+        # replace empty strings with nan so that they can be removed
+        self.logger.info(f"start cleaning")
+        empty_rows = []
+        def get_empty_rows(row):
+            if row["num_tokens"] <= self.minFeatureColLength:
+                empty_rows.append(row['__index_level_0__'])
```

Comment:
why don't you just set it to np.nan and then remove it with dropna?

Reply:
As far as I know dropna is not implemented for datasets (only for dataframes).